sparkly/utils.py

```diff
@@ -59,6 +64,89 @@ def absolute_path(file_path, *rel_path):
     )
 
 
+def chunks(seq, n):
+    """Iterate over seq by chunks of n elements.
+
+    >>> list(chunks([1, 2, 3, 4, 5], 2))
+    [[1, 2], [3, 4], [5]]
+    >>> list(chunks([], 2))
+    []
+    >>> list(chunks([1], 2))
+    [[1]]
+    """
+    it = iter(seq)
+    while True:
+        chunk = list(islice(it, n))
+        if not chunk:
+            break
+        yield chunk
+
+
+def data_type_defines(t, required_fields):
```

Comment:
Yeah, in the beginning I used such a name but then this function works for arbitrary types, not just `df.schema`s, so `dataType_has` or `dataType_defines` seemed better. Then I broke `dataType` to `data_type` because we don't really use camel case anywhere